File: camerax/lib/src/core/surface_oriented_metering_point_factory.dart
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;
import 'package:meta/meta.dart';

import 'metering_point_factory.dart';

final class SurfaceOrientedMeteringPointFactory extends MeteringPointFactory {
  final $interface.SurfaceOrientedMeteringPointFactory _obj;

  SurfaceOrientedMeteringPointFactory(double width, double height)
      : _obj = $interface.SurfaceOrientedMeteringPointFactory(width, height);

  @internal
  @override
  $interface.MeteringPointFactory get obj => _obj;
}

File: camerax_ios/ios/Classes/core/SurfaceOrientedMeteringPointFactory.swift
//
//  SurfaceOrientedMeteringPointFactory.swift
//  camerax_ios
//
//  Created by 闫守旺 on 2025/5/21.
//

import Foundation

public class SurfaceOrientedMeteringPointFactory: MeteringPointFactory {
    private let width: CGFloat
    private let height: CGFloat
    
    init(_ width: CGFloat, _ height: CGFloat) {
        self.width = width
        self.height = height
    }
    
    public func convertPoint(_ x: CGFloat, _ y: CGFloat) -> CGPoint {
        return CGPoint(x: x, y: y)
    }
}
